Windows: cri-dockerd keeps crashing

## Expected Behavior
cri-dockerd  runs normally

## Actual Behavior
cri-dockerd  keeps crashing
```logs
time="2024-03-08T15:02:13+08:00" level=info msg="Start cri-dockerd grpc backend"                                                    
panic: runtime error: invalid memory address or nil pointer dereference                                                             
[signal 0xc0000005 code=0x0 addr=0x18 pc=0x11baf06]                                                                                 
                                                                                                                                    
goroutine 49 [running]:                                                                                                             
k8s.io/cri-api/pkg/apis/runtime/v1.(*ContainerStats).MarshalToSizedBuffer(0x0, {0xc000189400, 0x4, 0x4f9})                          
        k8s.io/cri-api@v0.27.8/pkg/apis/runtime/v1/api.pb.go:17959 +0x26                                                            
k8s.io/cri-api/pkg/apis/runtime/v1.(*ListContainerStatsResponse).MarshalToSizedBuffer(0xc00070b760, {0xc000189400, 0x4f9, 0x4f9})   
        k8s.io/cri-api@v0.27.8/pkg/apis/runtime/v1/api.pb.go:17845 +0xaf                                                            
k8s.io/cri-api/pkg/apis/runtime/v1.(*ListContainerStatsResponse).Marshal(0xffffffffffffffff?)                                       
        k8s.io/cri-api@v0.27.8/pkg/apis/runtime/v1/api.pb.go:17825 +0x4d     
...
```

## Steps to Reproduce the Problem

  1. Set up kube on Windows Server node by installing Mirantis Kubernetes Engine (MKE) 3.6.11
  2. Deploy a k8s pod on Windows
  ```yaml
apiVersion: v1
kind: Pod
metadata:
  name: ping
  labels:
    app: ping
spec:
  containers:
  - name: ping
    image: mcr.microsoft.com/windows/nanoserver:ltsc2019
    imagePullPolicy: IfNotPresent
    command: ["ping", "-t", "127.0.0.1"]
  nodeSelector:
     kubernetes.io/os: "windows"
```
  3. Restart docker and `cri-dockerd` service
  4. Check `cri-dockerd` logs in `C:\k\cri-dockerd-mke\cri-dockerd-mke.log`

This issue also causes other weird issues, such as `kubectl logs -f` keeps aborted after a few seconds. E.g.
```
$ k logs ping --tail 10 -f :
...
Reply from 127.0.0.1: bytes=32 time<1ms TTL=128
Reply from 127.0.0.1: bytes=32 time<1ms TTL=128
rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing open //./pipe/cri-dockerd: The system cannot find the file specified.
```

## Specifications

  - Version: `cri-dockerd 0.3.9`
  - Platform: Windows Server 2019, docker 23.0.9
  - Subsystem: stats

--- a/core/stats.go
+++ b/core/stats.go
@@ -169,6 +169,9 @@ func (ds *dockerService) ContainerStats(
 	if err != nil {
 		return nil, err
 	}
+	if stats == nil {
+		return nil, fmt.Errorf("stats for container with id %s not available", r.ContainerId)
+	}
 	return &runtimeapi.ContainerStatsResponse{Stats: stats}, nil
 }
 
@@ -232,7 +235,9 @@ func (ds *dockerService) ListContainerStats(
 				return nil
 			}
 			mu.Lock()
-			results = append(results, stats)
+			if stats != nil {
+				results = append(results, stats)
+			}
 			mu.Unlock()
 			return nil
 		})

